spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -1,903 +1,589 @@
-        :server_web_idle_timeout                => 30000,
-        :server_connect_timeout                 => 120000,
-        :server_check_for_updates               => true,
-        :server_environment_class_cache_enabled => false,
-        :server_jruby9k                         => false,
-        :server_metrics                         => true,
-        :metrics_jmx_enable                     => true,
-        :metrics_graphite_enable                => true,
-        :metrics_graphite_host                  => 'graphitehost.example.com',
-        :metrics_graphite_port                  => 2003,
-        :metrics_server_id                      => 'puppetserver.example.com',
-        :metrics_graphite_interval              => 5,
-        :metrics_allowed                        => ['single.element.array'],
-        :server_experimental                    => true,
-        :server_ip                              => '0.0.0.0',
-        :server_port                            => '8140',
-        :server_http_port                       => '8139',
-        :server_ssl_ca_crl                      => '/etc/puppetlabs/puppet/ssl/ca/ca_crl.pem',
-        :server_ssl_ca_cert                     => '/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem',
-        :server_ssl_cert                        => '/etc/puppetlabs/puppet/ssl/certs/puppetserver123.example.com.pem',
-        :server_ssl_cert_key                    => '/etc/puppetlabs/puppet/ssl/private_keys/puppetserver123.example.com.pem',
-        :server_ssl_chain                       => '/etc/puppetlabs/puppet/ssl/ca/ca_crt.pem',
-        :server_crl_enable                      => true,
-        :server_trusted_agents                  => [],
-        :allow_header_cert_info                 => false,
-        :compile_mode                           => 'off', # In reality defaults to undef
-      } end
+      let(:auth_conf) { '/etc/custom/puppetserver/conf.d/auth.conf' }
+      let(:puppetserver_conf) { '/etc/custom/puppetserver/conf.d/puppetserver.conf' }
+
+      let(:params) do
+        {
+          server: true,
+          server_implementation: 'puppetserver',
+          # We set these values because they're calculated
+          server_jvm_config: '/etc/default/puppetserver',
+          server_jvm_min_heap_size: '2G',
+          server_jvm_max_heap_size: '2G',
+          server_jvm_extra_args: '',
+          server_max_active_instances: 2,
+          server_puppetserver_dir: '/etc/custom/puppetserver',
+          server_puppetserver_version: '2.4.99',
+        }
+      end
 
       describe 'with default parameters' do
-        let(:params) do
-          default_params.merge(:server_puppetserver_dir => '/etc/custom/puppetserver')
-        end
         it { should contain_file('/etc/custom/puppetserver/bootstrap.cfg') }
-        it { should contain_file_line('ca_enabled').with_ensure('present') }
-        it { should contain_file_line('ca_disabled'). with_ensure('absent') }
```

Comment:
What's the reason for removing these two ensures?

Reply:
Oh, that was my bad. Initially I changed this to `should_not` because I removed the version pinning as well but then you test with the latest version which doesn't have them. Will add them again.